bin/replicate-postgresql.sh

```diff
@@ -66,16 +76,9 @@ until govuk-docker run "$postgres_container" /usr/bin/psql -h "$postgres_contain
   sleep 1
 done
 
-database="$app"
+# Extract the local database name from the app's DATABASE_URL environment variable
+database="$(govuk-docker config | ruby -ryaml -e "puts YAML::load(STDIN.read).dig('services', '${app}-app', 'environment', 'DATABASE_URL').split('/').last")"
 
 govuk-docker run "$postgres_container" /usr/bin/psql -h "$postgres_container" -U postgres -c "DROP DATABASE IF EXISTS \"${database}\""
 govuk-docker run "$postgres_container" /usr/bin/createdb -h "$postgres_container" -U postgres "$database"
-if [[ "$postgres_container" != "postgres-9.6" ]]; then
-  # Restoring a dump created with postgres 9.6 into postgres 13 gives
-  # an error about the "public" schema already existing.  The SQL dump
-  # includes a CREATE SCHEMA line, so I don't know why it's fine with
-  # older postgres.  We can drop this conditional and line when we're
-  # only replicating data dumped with a newer pg_dump.
-  govuk-docker run "$postgres_container" /usr/bin/psql -h "$postgres_container" -U postgres -d "$database" -c 'DROP SCHEMA IF EXISTS "public"'
-fi
-pv "$archive_path" | govuk-docker run "$postgres_container" /usr/bin/pg_restore -h "$postgres_container" -U postgres -d "$database" --no-owner
+pv "$archive_path" | govuk-docker run "$postgres_container" /usr/bin/pg_restore -h "$postgres_container" -U postgres -d "$database" --no-owner --no-privileges
```

Comment:
It took me a minute to figure out what had changed about this line. Can you explain why `--no-privileges` was added? Should it maybe be encapsulated in its own commit?

Reply:
Yes – this probably should be its own commit. I'll split it out. Thanks for the nudge! 😄

When restoring Postgres 13 dumps into a Postgres 13 docker container, I was getting an error:

```
pg_restore: error: could not execute query: ERROR: role "rdsadmin" does not exist
```

I believe `pg_restore` was trying to reinstate some permissions on an RDS-specific user role which doesn't exist in the Docker environment. I think this must be a new behaviour (possibly something new in RDS) because it doesn't look like this was necessary with Postgres 9.6.

Essentially, our interest is in importing the database schema and data – not the various user roles and permissions that exist in RDS.

Just to be clear, this seems to be independent of the conditional `DROP SCHEMA IF EXISTS "public"` that was needed when moving between Postgres versions.